### PR TITLE
PROVISION: Split Access key into Get All Labs and Get Available Labs access keys

### PR DESCRIPTION
**Pull request by geetika-jain**

### CODE REVIEW

File: DMX.Core.Api/Brokers/LabApis/ExternalLabApiBroker.cs [hunk 1]
```diff
@@ -42,12 +44,21 @@ private IRESTFulApiFactoryClient GetApiClient(IConfiguration configuration)
...
+        private static string GetAvailableDevicesAccessToken(IConfiguration configuration)
         {
             LocalConfigurations localConfigurations =
                 configuration.Get<LocalConfigurations>();
 
-            return localConfigurations.ApiConfigurations.AccessKey;
+            return localConfigurations.ApiConfigurations.GetAvailableDevicesAccessKey;
+        }
+
+
```
Comment (hassanhabib) [inline DMX.Core.Api/Brokers/LabApis/ExternalLabApiBroker.cs:55]: Extra line here

File: DMX.Core.Api.Infrastructure.Build/Services/ScriptGenerations/ScriptGenerationService.cs [hunk 2]
```diff
@@ -124,7 +124,8 @@ public void GenerateProvisionScript()
                             { "AzureSqlServerAdminName", "${{ secrets.AZURE_SQLSERVER_ADMINNAME }}" },
                             { "AzureSqlServerAdminAccess", "${{ secrets.AZURE_SQLSERVER_ADMINACCESS }}" },
                             { "AzureAppServiceExternalLabApiUrl", "${{ secrets.AZURE_APPSERVICE_EXTERNALLABAPI_URL }}" },
-                            { "AzureAppServiceExternalLabApiAccessKey", "${{ secrets.AZURE_APPSERVICE_EXTERNALLABAPI_ACCESSKEY }}" }
+                            { "AzureAppServiceExternalLabApiGetAllDevicesAccessKey", "${{ secrets.AZURE_APPSERVICE_EXTERNALLABAPI_GETALLDEVICES_ACCESSKEY }}" },
```
Comment (hassanhabib) [inline DMX.Core.Api.Infrastructure.Build/Services/ScriptGenerations/ScriptGenerationService.cs:127]: This line and the one below it over > 120 chars. split.

File: DMX.Core.Api/Models/Configurations/ApiConfigurations.cs [hunk 3]
```diff
@@ -6,7 +6,8 @@ namespace DMX.Core.Api.Models.Configurations
 {
     public class ApiConfigurations
     {
-        public string Url { get; set; }
-        public string AccessKey { get; set; }
+        public string Url { get; set; }
+        public string GetAllDevicesAccessKey { get; internal set; }
+        public string GetAvailableDevicesAccessKey { get; internal set; }
```
Comment (hassanhabib) [inline DMX.Core.Api/Models/Configurations/ApiConfigurations.cs:11]: Change these property names from verb to nouns and remove the `internal` part.

File: DMX.Core.Api/Brokers/LabApis/IExternalLabApiBroker.Labs.cs [hunk 4]
```diff
@@ -10,5 +10,6 @@ namespace DMX.Core.Api.Brokers.LabApis
     public partial interface IExternalLabApiBroker
     {
         ValueTask<ExternalLabCollection> GetAvailableLabsAsync();
+        ValueTask<ExternalLabCollection> GetAllLabsAsync();
```
Comment (hassanhabib) [inline DMX.Core.Api/Brokers/LabApis/IExternalLabApiBroker.Labs.cs:13]: Make this method above - L comes before V

File: DMX.Core.Api/Models/Configurations/ApiConfigurations.cs (hunk 3 above)
Comment (mimendel) [reply DMX.Core.Api/Models/Configurations/ApiConfigurations.cs]: I changed the property names, and assumed that the CloudBroker.WebApps variable should also be updated. Is that correct?

**Comment by hassanhabib** [comment]
@geetika-jain conflict here.

**Comment by geetika-jain** [comment]
Resolved. waiting for build to finish

File: DMX.Core.Api.Infrastructure.Build/Services/ScriptGenerations/ScriptGenerationService.cs [hunk 5]
```diff
@@ -118,15 +118,29 @@ public void GenerateProvisionScript()
...
+                            { "AzureAdAppDmxCoreScopesGetLabCommand", 
+                                "${{ secrets.AZURE_ADAPP_DMXCORE_SCOPES_GETLABCOMMAND }}" },
+
                             { "AzureSqlServerAdminName", "${{ secrets.AZURE_SQLSERVER_ADMINNAME }}" },
                             { "AzureSqlServerAdminAccess", "${{ secrets.AZURE_SQLSERVER_ADMINACCESS }}" },
-                            { "AzureAppServiceExternalLabApiUrl", "${{ secrets.AZURE_APPSERVICE_EXTERNALLABAPI_URL }}" },
-                            { "AzureAppServiceExternalLabApiAccessKey", "${{ secrets.AZURE_APPSERVICE_EXTERNALLABAPI_ACCESSKEY }}" }
+
+                            { "AzureAppServiceExternalLabApiUrl",
+                                "${{ secrets.AZURE_APPSERVICE_EXTERNALLABAPI_URL }}" },
```
Comment (hassanhabib) [inline DMX.Core.Api.Infrastructure.Build/Services/ScriptGenerations/ScriptGenerationService.cs:140]: Put them right under each other. no tabs on the second line

Comment (hassanhabib) [reply DMX.Core.Api.Infrastructure.Build/Services/ScriptGenerations/ScriptGenerationService.cs:140]: Do the same for the rest.

**Comment by terrypalo** [comment]
The provisioning build broke the instance. We should make sure to add the correct secrets within GitHub before making any changes to the Provisioning project.

**Review by hassanhabib — APPROVED** [review]

**Pull request merged**